keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -1172,6 +1207,179 @@ back to aggregating container requests.
+
+###### Tracking Actual Pod-level Resources
+To accurately track actual pod-level resources during in-place pod resizing, several
+changes are required:
+
+1. Configuration reading: Pod-level resource config is currently read as part of the
+   resize flow, but will also need to be read during pod creation. Critically, the
+   configuration must be read again after the resize operation to capture the
+   updated resource values.  Currently, the configuration is only read before a
+   resize.
+   
+2. Pod Status Update: Because the pod status is updated before the resize takes
+   effect, the status will not immediately reflect the new resource values.  If a
+   container within the pod is also being resized, the container resize operation
+   will trigger a pod synchronization (pod-sync), which will refresh the pod's
+   status.  However, if only pod-level resources are being resized, a pod-sync must
+   be explicitly triggered to update the pod status with the new resource
+   allocation.
+
+3. [Scoped for Beta] Caching: Actual pod resource data may be cached to minimize API server load. This cache, if implemented, must be invalidated after each successful pod resize to ensure that subsequent reads retrieve the latest information. The need for and implementation of this caching mechanism will be evaluated in the beta phase.  Performance benchmarking will be conducted to determine if caching is required and, if so, what caching strategy is most appropriate.
+
+**Note for future enhancements for Ephemeral containers with pod-level resources and
+IPPR**
+Previously, assigning resources to ephemeral
+containers wasn't allowed because pod resource allocations were immutable. With
+the introduction of in-place pod resizing, users will gain more flexibility:
+
+* Adjust pod-level resources to accommodate the needs of ephemeral containers. This
+allows for a more dynamic allocation of resources within the pod.
+* Specify resource requests and limits directly for ephemeral containers. Kubernetes will
+then automatically resize the pod to ensure sufficient resources are available
+for both regular and ephemeral containers.
+
+Currently, setting `resources` for ephemeral containers is disallowed as pod
+resource allocations were immutable before In-Place Pod Resizing feature. With
+in-place pod resize for pod-level resource allocation, users should be able to
+either modify the pod-level resources to accommodate ephemeral containers or
+supply resources at container-level for ephemeral containers and kubernetes will
+resize the pod to accommodate the ephemeral containers.
+
+#### [Scoped for Beta] Topology Manager
+
+Note: This section includes only high level overview; Design details will be added in Beta stage.
+
+* The pod level scope for topology aligntment will consider pod level requests and limits instead of container level aggregates.
+* The hint providers will consider pod level requests and limits instead of
+  container level aggregates.
```

Comment:
How do we handle the current use case though? if the user does _not_ set pod-level resources, will we have 2 code paths (the existing + the new one being proposed), or do we want to automatically compute the pod-level resources aggregating the container-level resources, improving on what the code currently does?

For topology manager we will probably need some more details about the design

Reply:
- When pod-level resources are not specified, as in the current use case, we change nothing in the pod spec. Today, cgroups at pod-level are still calculated using the aggregation method. We will continue to have that path for pod spec without pod level resources. So to summarize, pod-level resources are calculated even today and used in cgroup values at pod-level, but not updated in pod-spec. 

- We definitely need more details around design for topology manager which is why I intentionally want to skip it for alpha and focus on CPU manager and memory manager first for alpha.

Reply:
What about the case where pod level requests/limits are specified but `container` scope configured on the node? Do we disregard pod level requests/limits in that case?

Reply:
Additionally, It might be obvious, but it's important to emphasize that in case pod level resources are specified we must consider the topology manager's policies and policy options configured on the node.